cg/server/endpoints/samples.py

```diff
@@ -46,21 +44,30 @@ def parse_sample_in_collaboration(sample_id):
 @SAMPLES_BLUEPRINT.route("/samples")
 def get_samples():
     """Return samples."""
-    if request.args.get("status") and not g.current_user.is_admin:
-        return abort(HTTPStatus.FORBIDDEN)
-    if request.args.get("status") == "incoming":
+    samples_request = SamplesRequest.model_validate(request.args.to_dict())
+    if samples_request.status:
+        if not g.current_user.is_admin:
+            return abort(HTTPStatus.FORBIDDEN)
+        else:
+            return _get_samples_handled_from_lims(request=samples_request)
+    customers: list[Customer] | None = None if g.current_user.is_admin else g.current_user.customers
+    samples, total = db.get_samples_by_customers_and_pattern(
+        pattern=samples_request.enquiry, customers=customers
+    )
+    parsed_samples: list[dict] = [sample.to_dict() for sample in samples]
+    return jsonify(samples=parsed_samples, total=total)
+
+
+def _get_samples_handled_from_lims(request: SamplesRequest):
+    """Get samples based on the provided status."""
+    if request.status == "incoming":
         samples: list[Sample] = db.get_samples_to_receive()
-    elif request.args.get("status") == "labprep":
+    elif request.status == "labprep":
         samples: list[Sample] = db.get_samples_to_prepare()
-    elif request.args.get("status") == "sequencing":
+    elif request.status == "sequencing":
         samples: list[Sample] = db.get_samples_to_sequence()
     else:
-        customers: list[Customer] | None = (
-            None if g.current_user.is_admin else g.current_user.customers
-        )
-        samples: list[Sample] = db.get_samples_by_customers_and_pattern(
-            pattern=request.args.get("enquiry"), customers=customers
-        )
-    limit = int(request.args.get("limit", 50))
+        return abort(HTTPStatus.BAD_REQUEST)
+    limit: int = request.page_size or 50
```

Comment:
Have a look at the `cg/services/sample_service/sample_service.py`

I would decouple this endpoint from the db as much as possible and let the service handle the db interaction. 
There is also a function called `create_samples_response` that has a sample_dto. 

If you consider this out of scope then ignore. However, since most of it is already setup time required might be minimal for a major improvement.

Reply:
moving to sample service